clang/test/SemaSYCL/kernel-assert.cpp

```diff
@@ -0,0 +1,21 @@
+// REQUIRES: system-windows
+// RUN: %clang_cc1 -fsycl-is-device -triple nvptx64-nvidia-cuda -internal-isystem %S/Inputs/ -fsyntax-only -verify %s
+
+// expected-no-diagnostics
+
+#include <assert.h>
+
+// Make sure that SYCL kernels can correctly call asserts on Windows. We do not
+// expect to see any warnings or compilation errors. On Windows targeting NVPTX
+// the implementation of assert is provided by libclc.
+//
+// Please note, that the execution of the assert is tested in llvm-test-suite.
```

Comment:
If this functionality is being tested in the test suite and we no longer have FE changes, I don't think we need FE tests. Can you explain why you added this test? I'm not sure if I am missing something.